frontend/src/components/Admin/DevPortfolio/DevPortfolioDetails.tsx

```diff
@@ -8,13 +10,64 @@ type Props = {
   isAdminView: boolean;
 };
 
+const sortSubmissions = (submissions: DevPortfolioSubmission[]) =>
+  submissions.sort((s1, s2) => s1.member.netid.localeCompare(s2.member.netid));
+
 const DevPortfolioDetails: React.FC<Props> = ({ uuid, isAdminView }) => {
   const [portfolio, setPortfolio] = useState<DevPortfolio | null>(null);
 
   useEffect(() => {
     DevPortfolioAPI.getDevPortfolio(uuid, isAdminView).then((portfolio) => setPortfolio(portfolio));
   }, [uuid, isAdminView]);
 
+  const handleExportToCsv = () => {
+    if (portfolio?.submissions === undefined || portfolio?.submissions.length <= 0) {
+      Emitters.generalError.emit({
+        headerMsg: 'Failed to export Dev Portfolio Submissions to CSV',
+        contentMsg: 'Please make sure there is at least 1 submission in the table.'
+      });
+      return;
+    }
+
+    const uniqueIds: string[] = [];
+
+    const uniqueSubmissions = portfolio.submissions.filter((submission) => {
+      const isDuplicate = uniqueIds.includes(submission.member.netid);
+      if (!isDuplicate) {
+        uniqueIds.push(submission.member.netid);
+        return true;
+      }
+      return false;
+    });
+
+    const sortedSubmissions = sortSubmissions(uniqueSubmissions);
+
+    const csvData = sortedSubmissions.map((submission) => ({
+      name: `${submission.member.firstName} ${submission.member.lastName}`,
+      netid: submission.member.netid,
+      opened_score: Number(submission.openedPRs.some((pr) => pr.status === 'valid')),
+      reviewed_score: Number(submission.reviewedPRs.some((pr) => pr.status === 'valid')),
+      total_score:
+        Number(submission.reviewedPRs.some((pr) => pr.status === 'valid')) +
+        Number(submission.openedPRs.some((pr) => pr.status === 'valid'))
```

Comment:
The opened score and reviewed score only needs to be computed to once and can be stored in a var probably